src/main/java/me/zeroeightsix/kami/module/modules/player/ViewLock.kt

```diff
@@ -0,0 +1,114 @@
+    private val yawSlice = register(Settings.integerBuilder("YawSlice").withValue(8).withRange(2, 32).withStep(1).withVisibility { autoYaw.value })
+    private val pitchSlice = register(Settings.integerBuilder("PitchSlice").withValue(5).withRange(2, 32).withStep(1).withVisibility { autoPitch.value })
+
+    private var yawSanp = 0
+    private var pitchSnap = 0
+
+
+    override fun onEnable() {
+        yawSliceAngle = 360.0f / yawSlice.value
+        pitchSliceAngle = 180.0f / (pitchSlice.value - 1)
+        if ((autoYaw.value || autoPitch.value) && mc.player != null)
+            snapToNext()
+    }
+
+    init {
+        listener<SafeTickEvent> {
+            if (autoYaw.value || autoPitch.value) {
+                snapToSlice()
+
+            } else {
+                if (yaw.value) mc.player.rotationYaw = specifcYaw.value
+                if (pitch.value) mc.player.rotationPitch = specificPitch.value
+            }
+        }
+        yawSlice.settingListener = Setting.SettingListeners {
+            if (isEnabled && autoYaw.value) snapToNext()
+            yawSliceAngle = 360.0f / yawSlice.value
+        }
+        pitchSlice.settingListener = Setting.SettingListeners {
+            if (isEnabled && autoPitch.value) snapToNext()
+            pitchSliceAngle = 180.0f / (pitchSlice.value - 1)
+        }
+        autoPitch.settingListener = Setting.SettingListeners { if (isEnabled && autoPitch.value) snapToNext() }
+        autoYaw.settingListener = Setting.SettingListeners { if (isEnabled && autoYaw.value) snapToNext() }
+    }
+
+    private fun snapToNext() {
+        yawSanp = round(mc.player.rotationYaw / yawSliceAngle).toInt()
+        pitchSnap = round((mc.player.rotationPitch + 90) / pitchSliceAngle).toInt()
+        snapToSlice()
+    }
+
+    private fun changeDirection(yawChange: Int, pitchChange: Int) {
+        yawSanp = Math.floorMod(yawSanp + yawChange, yawSlice.value)
+        pitchSnap = min(max(pitchSnap + pitchChange, 0), pitchSlice.value - 1)
+        snapToSlice()
+    }
+
+    private fun snapToSlice() {
+        if (yaw.value && autoYaw.value) {
+            mc.player.rotationYaw = yawSanp * yawSliceAngle
+            mc.player.ridingEntity?.let { it.rotationYaw = mc.player.rotationYaw }
+        }
+        if (pitch.value && autoPitch.value) mc.player.rotationPitch = pitchSnap * pitchSliceAngle - 90
+    }
+
+
+    fun handleDeltaX(deltaX: Int) {
+        val currenttime = System.currentTimeMillis()
+        deltaXQueue.add(Pair(deltaX, currenttime))
+        val sum = deltaXQueue.sumBy { it.first }
+        println(deltaXQueue.size)
+        if (abs(sum) > threshold.value * 100) {
+            deltaXQueue.clear()
+            changeDirection(sign(sum.toDouble()).toInt(), 0)
+            return
+        }
+        while (deltaXQueue.peek().second < currenttime - 500)
+            deltaXQueue.remove()
```

Comment:
Instead of using a while loop to check. Use `.removeIf()` would be more simple

Reply:
remove if will run over the entire list, which is not necessary as we know that they are ordered by timestamp